java/src/main/java/io/process/analytics/tools/bpmn/generator/converter/AlgoToDisplayModelConverter.java

```diff
@@ -78,11 +80,187 @@ public DisplayModel convert(Grid grid, Diagram diagram) {
+                wayPoints.add(new DisplayPoint(dimensionTo.x  + dimensionTo.width / 2, dimensionTo.y));
+                break;
+            case TopLeftToBottomRight_FirstVertical: // vertical then horizontal
+                wayPoints.add(new DisplayPoint(dimensionFrom.x + dimensionFrom.width / 2, dimensionFrom.y + dimensionTo.height));
+                wayPoints.add(new DisplayPoint(dimensionFrom.x + dimensionFrom.width / 2, dimensionTo.y + dimensionTo.height / 2));
+                wayPoints.add(new DisplayPoint(dimensionTo.x, dimensionTo.y + dimensionTo.height / 2));
+                break;
+            case TopRightToBottomLeft_FirstHorizontal: // horizontal then vertical
+                wayPoints.add(new DisplayPoint(dimensionFrom.x, dimensionFrom.y + dimensionFrom.height / 2));
+                wayPoints.add(new DisplayPoint(dimensionTo.x + dimensionTo.width / 2, dimensionFrom.y + dimensionFrom.height / 2));
+                wayPoints.add(new DisplayPoint(dimensionTo.x  + dimensionTo.width / 2, dimensionTo.y));
+                break;
+            case TopRightToBottomLeft_FirstVertical: // vertical then horizontal
+                wayPoints.add(new DisplayPoint(dimensionFrom.x + dimensionFrom.width / 2, dimensionFrom.y + dimensionTo.height));
+                wayPoints.add(new DisplayPoint(dimensionFrom.x + dimensionFrom.width / 2, dimensionTo.y + dimensionTo.height / 2));
+                wayPoints.add(new DisplayPoint(dimensionTo.x + dimensionTo.width / 2, dimensionTo.y + dimensionTo.height / 2));
+                break;
+            case VerticalBottomToTop:
+                wayPoints.add(new DisplayPoint(dimensionFrom.x + dimensionFrom.width / 2, dimensionFrom.y));
+                wayPoints.add(new DisplayPoint(dimensionTo.x  + dimensionTo.width / 2, dimensionTo.y + dimensionTo.height));
+                break;
+            case VerticalTopToBottom:
+                wayPoints.add(new DisplayPoint(dimensionFrom.x + dimensionFrom.width / 2, dimensionFrom.y + dimensionFrom.height));
+                wayPoints.add(new DisplayPoint(dimensionTo.x  + dimensionTo.width / 2, dimensionTo.y));
+                break;
+            default:
+                // do nothing
+        }
+        return wayPoints;
+    }
+
+    private DisplayFlowNode getFlowNode(List<DisplayFlowNode> flowNodes, String flowNodeId) {
+        return flowNodes.stream()
+                .filter(f -> flowNodeId.equals(f.bpmnElementId))
+                .findFirst()
+                .get(); // always exist, otherwise error occur on flow node generation
+    }
+
+    private Position getPositionOfShape(Grid grid, String shapeId) {
+        return grid.getPositions()
+                .stream()
+                .filter(p -> shapeId.equals(p.getShape()))
+                .findFirst()
+                .get(); // always exist, otherwise error occur on flow node generation
+    }
+
+    // visible for testing
```

Comment:
ℹ️  IMO this method is static and visible for testing because it should be extracted in a specific class